xbmc/utils/test/TestSysfsUtils.cpp

```diff
@@ -0,0 +1,46 @@
+/*
+ *  Copyright (C) 2005-2018 Team Kodi
+ *  This file is part of Kodi - https://kodi.tv
+ *
+ *  SPDX-License-Identifier: GPL-2.0-or-later
+ *  See LICENSES/README.md for more information.
+ */
+
+#include "utils/SysfsUtils.h"
+
+#include "gtest/gtest.h"
+
+#include <cstdio>
+#include <fstream>
+#include <string>
+
+TEST(TestSysfsUtils, String)
+{
+  std::string path("/tmp/kodi-test");
+  std::ofstream output(path);
+
+  ASSERT_TRUE(SysfsUtils::SetString(path, "test"));
+  ASSERT_TRUE(SysfsUtils::GetString(path) == "test");
+
+  std::remove(path.c_str());
```

Comment:
Consider moving the removal to a `TearDown` method in a test fixture. At the moment if either assertion fails, we will never get to this line.

Reply:
e.g.:

```c++
struct TestSysfsUtils : public ::testing::Test
{
  TestSysfsUtils()
   : path("/tmp/kodi-test")
   , output(path)
  {
  }

  ~TestSysfsUtils()
  {
    std::remove(path.c_str());
  }
  
  std::string path;
  std::ofstream output;
};

TEST_F(TestSysfsUtils, String)
{
  ASSERT_TRUE(SysfsUtils::SetString(path, "test"));
  ASSERT_TRUE(SysfsUtils::GetString(path) == "test");
}

TEST_F(TestSysfsUtils, Int)
{
  ASSERT_TRUE(SysfsUtils::SetInt(path, 1234));
  ASSERT_TRUE(SysfsUtils::GetInt(path) == 1234);
}
```

Reply:
Thanks for the tip! That helps a lot! :100: